### PR TITLE
Add cloudbuild config to build gcp-compute-persistent-disk-csi-driver image

### PR DESCRIPTION
**Pull request by cpanato**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

As part of the effort described in this [issue](https://github.com/kubernetes/k8s.io/issues/1525) we are deprecating the `gcr.io/gke-release` and the image `gcr.io/gke-release/gcp-compute-persistent-disk-csi-driver` does not exist in other registries. So we need to build that and push to a new registry.

We will use `gcr.io/k8s-staging-cloud-provider-gcp` to push the images and This PR add the cloudbuild to generate and push the image.

for more context see the slack 🧵  https://kubernetes.slack.com/archives/C09QZFCE5/p1615563067038600bump

- Also bump the go version to the lastest 1.13 available

`k/test-infra` PR to add the prow job: TBC

/assign @spiffxp @msau42 @mattcary 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add cloudbuild config to build gcp-compute-persistent-disk-csi-driver image
```

### CODE REVIEW

**Comment by mattcary** [comment]
Note the integration failure is due to #721 for which a fix is on its way.

File: cloudbuild.yaml
Comment (mattcary) [inline cloudbuild.yaml:9]: Can't we just use make as the entrypoint? See eg https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/cloudbuild.yaml.

File: cloudbuild.yaml
Comment (mattcary) [inline cloudbuild.yaml:23]: How does this registry work?

In particular I think in the filestore driver we seem to just get it from the project (q.v.)

Comment (cpanato) [reply cloudbuild.yaml:23]: this is set in the prow job, to configure the registry name like `gcr.io/my-registry` we did that in the k/release to support other registries in the cloudbuild for any other users that are using that. But we can make it static in this case. I don't have a strong opinion on that.

File: cloudbuild.yaml
Comment (cpanato) [reply cloudbuild.yaml:9]: sounds good, will update

File: cloudbuild.yaml
Comment (cpanato) [reply cloudbuild.yaml:23]: reverted and make similar in what we have for gcp-filestore-csi-driver

**Comment by cpanato** [comment]
testing this cloudbuild on my GCP account, and everything worked fine

**Comment by mattcary** [comment]
/lgtm

thanks tons for helping with this!

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by spiffxp** [comment]
/assign @mattcary @msau42 
For the /approve, I don't have rights here

**Comment by mattcary** [comment]
Oops I can never keep track of when I have to lgtm and when I have to approve. Thanks @spiffxp 

/lgtm
/approve

**Comment by cpanato** [comment]
job ran successfully (https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-gcp-compute-persistent-disk-csi-driver-push-images/1371915861887029248)  and the master tag is available: `docker pull gcr.io/k8s-staging-cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver:master`

**Pull request merged**
